holoviews/core/data/__init__.py

```diff
@@ -316,7 +316,10 @@ def select(self, selection_specs=None, **selection):
         supplied, which will ensure the selection is only applied if the
         specs match the selected object.
         """
-        if selection_specs and not any(self.matches(sp) for sp in selection_specs):
+        selection = {dim: sel for dim, sel in selection.items()
+                     if dim in self.dimensions()+['selection_mask']}
```

Comment:
It's used a bunch in the interfaces and allows applying a boolean mask in ``__getitem__``.